scripts/pre_commit_linter.py

```diff
@@ -1049,6 +1049,59 @@ def _check_docstrings(all_files):
     return summary_messages
 
 
+def _check_controller_class_names(all_files):
+    """This function checks that all controller class names end with
+    either 'Handler' or 'Page'.
+    """
+    print 'Starting controller class names check...'
+    print '----------------------------------------'
+    summary_messages = []
+    controllers_path = 'core/controllers*'
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.py') and not filename.endswith('_test.py')
+        and fnmatch.fnmatch(filename, controllers_path)]
+    message = (
+        'Please ensure that the name of this class ends with \'Handler\' '
+        'or \'Page\'.')
+    failed = False
+    for filename in files_to_check:
+        with open(filename, 'r') as f:
+            file_content = f.readlines()
+            file_length = len(file_content)
+            for line_num in range(file_length):
+                line = file_content[line_num].lstrip().rstrip()
+                if line.startswith('class '):
+                    # Check that the name of the class ends with 'Handler'
+                    # or 'Page'.
+                    class_name = line.split(' ')[1].split('(')[0]
+                    super_class_name = line.split('(')[1][:-2]
+                    if (super_class_name.endswith('Handler') and
+                            not class_name.endswith('Handler')
+                            and not class_name.endswith('Page')):
```

Comment:
I think you'll have to change back some of the class names which render an HTML page to end with `Page` and not with `Handler`. @seanlip: can you confirm this please? Thanks!

Reply:
That is correct. You can probably even test this: make sure that any class ending with `Page` has a single "get()" method that returns HTML, and that all such classes end with 'Page'. Any other class should end with 'Handler'.

Reply:
@seanlip @apb7 done!